src/util.js

```diff
@@ -44,7 +44,7 @@ export const recursiveMerge = (
     // TODO(jlfwong): Handle malformed input where a and b are not the same
     // type.
 
-    if (typeof a !== 'object') {
+    if (typeof a !== 'object' || Array.isArray(b)) {
```

Comment:
@xymostech Sorry, just re-read your comment and realised I completely missed the point. So do you think the code should be something like:
```
if (typeof a !== 'object' || Array.isArray(a) || Array.isArray(b)) {
```
Actually I think we probably need to take it further
```
if (typeof a !== typeof b && typeof a !== 'object'
  || Array.isArray(a)
  || Array.isArray(b)
  || a === null
  || b === null
) {
```
as this would then cover off the situation where we want to merge `{a:  {b: 2}}` and `{a: 2}` and when we're merging `{a: 1}` with `{a: null}` ...although it doesn't look pretty.

Let me know your thoughts